### PR TITLE
Convert keyword elements correctly

### PR DESCRIPTION
**Pull request by MattiasRosell**
I noticed that the `<del>` tag was not converted into htpy's `del_` correctly. The formatters also can not handle importing python's `del` keyword and would crash. Ruff would not output anything while Black would output the unformatted code.

I fixed these issues by: 

- Replacing `div` with `div_` when converting html using html2htpy.
- Adding a test checking that `div` is successfully converted to `div_`.
- Adding consistent behaviour between Ruff and Black when the input can not be formatted.

@pelme then found out that htpy did not convert the `del_` element to html correctly. So we fixed that as well.

- Fix the `del_` element not converting into `<del>` correctly.
- Only remove underscore from elements named after Python's keywords.

### CODE REVIEW

File: tests/test_html2htpy.py [hunk 1]
```diff
@@ -402,3 +402,13 @@ def test_shorthand_contains_dot() -> None:
...
+
+
+def test_del_tag_is_replaced_with_del_() -> None:
+    actual = html2htpy(
+        "<div><del>deleted</del></div>",
+        shorthand_id_class=True,
+        import_mode="yes",
+    )
+
+    assert actual == 'from htpy import del_, div\ndiv[del_["deleted"]]'
```
Comment (pelme) [inline tests/test_html2htpy.py:414]: issue: htpy itself does not handle the trailing underscore properly. 😆  Did not consider that before.

```py
>>> from htpy import del_
>>> print(del_)
<del_></del_>
```


suggestion: We should also deal with the trailing underscores in the element names.  We need to do `name.removesuffix("_")` in `htpy/__init__.py` / `_get_element()` to remove the trailing `_`.

File: htpy/html2htpy.py [hunk 2]
```diff
@@ -147,6 +147,9 @@ def __init__(
 
     @property
     def python_element_name(self) -> str:
+        # Note: <del> tags should be translated to "del_" instead of "del" in htpy.
+        if self.html_tag == "del":
```
Comment (pelme) [inline htpy/html2htpy.py:151]:
```suggestion
        if keyword.iskeyword(self.html_tag):
            return self.html_tag + "_"
```

I guess `del` is the only builtin html element that clashes with python keywords. But keyword.iskeyword() is a more general approach that avoids the problem altogether now+in the future if new keywords are added to python. we use the same approach for attributes in this file so the keyword module is already imported: https://docs.python.org/3/library/keyword.html

**Review by pelme — CHANGES_REQUESTED** [review]
praise: very nice, fixing annoyances in html2htpy is really helpful and makes it nicer to use!

File: htpy/html2htpy.py (hunk 2 above)
Comment (MattiasRosell) [reply htpy/html2htpy.py]: I agree! That is a lot cleaner! Fixed!

File: tests/test_element.py [hunk 3]
```diff
@@ -42,3 +42,7 @@ def test_element_type() -> None:
 
     assert_type(div()["a"], Element)
     assert isinstance(div()["a"], Element)
+
+
+def test_del_keyword_element() -> None:
+    assert str(del_) == "<del></del>"
```
Comment (pelme) [inline tests/test_element.py:48]: could you also add a test for an element that is not part of the predefined list? continue? class or similar?

Comment (MattiasRosell) [reply tests/test_element.py]: Done!

File: tests/test_html2htpy.py (hunk 1 above)
Comment (MattiasRosell) [reply tests/test_html2htpy.py:414]: Done!

File: tests/test_element.py [hunk 4]
```diff
@@ -42,3 +42,21 @@ def test_element_type() -> None:
...
+    [(del_, "<del></del>"), (class_, "<class></class>"), (break_, "<break></break>")],
+)
+def test_keyword_named_elements(element: Element, expected: str) -> None:
+    actual = str(element)
+    assert actual == expected
+
+
+@pytest.mark.parametrize(
+    ("element", "expected"),
+    [(imaginary, "<imaginary></imaginary>"), (imaginary_, "<imaginary-></imaginary->")],
```
Comment (MattiasRosell) [inline tests/test_element.py:58]: Should underscore always be replaced with a hyphen like this?

**Review by MattiasRosell — COMMENTED** [review]
Updated branch to fix conflicts

File: htpy/__init__.py [hunk 5]
```diff
@@ -74,12 +76,18 @@ def _id_class_names_from_css_str(x: t.Any) -> dict[str, Attribute]:
     return result
 
 
-def _kwarg_attribute_name(name: str) -> str:
+def _html_name(name: str) -> str:
```
Comment (pelme) [inline htpy/__init__.py:79]:
```suggestion
def _python_to_html_name(name: str) -> str:
```

suggestion: maybe it is clearer?

File: htpy/html2htpy.py [hunk 6]
```diff
@@ -160,28 +162,40 @@ def serialize(self, *, shorthand_id_class: bool, use_h_prefix: bool) -> str:
...
+
     def format(self, s: str) -> str:
         result = subprocess.run(
             ["ruff", "format", "-"],
             input=s.encode("utf8"),
             stdout=subprocess.PIPE,
         )
+        if result.returncode == self.error_return_code:
+            _printerr("Ruff failed to parse the input. The output will be left unformatted.")
+            return s
```
Comment (pelme) [inline htpy/html2htpy.py:198]: question: when does this happen? is this not always a bug in html2htpy? if it produces invalid python code? 😬 

suggestion: can we clarify this and add something like this to the error message: "This is likely a bug in html2htpy. Please report this as an issue to htpy: https://github.com/pelme/htpy/issues".

File: tests/test_element.py (hunk 4 above)
Comment (pelme) [reply tests/test_element.py:71]: I cannot thing of much other useful things to do. Another option would be to raise an error for trailing underscores for non keywords but... it feels to harsh too. 😆

**Review by pelme — COMMENTED** [review]
praise: very nice, having a html2htpy that works reliable is nice!

**Pull request merged**